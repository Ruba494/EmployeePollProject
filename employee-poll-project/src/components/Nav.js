import React from 'react'
import {Link} from 'react-router-dom'
import LogOut from './LogOut'
import { connect } from "react-redux";


export const Nav = (props) => {

    return <nav className='nav'>
        <ul>
            <li key='home'>
                <Link to='/'>Home</Link>
            </li>
            <li key='leadBoard'>
                <Link to='/leaderboard'>Lead Board</Link>
            </li>
            <li key='new'>
                <Link to='/add'>Lead Board</Link>
            </li>
        </ul>
        <div>
        {props.authedUser === null  ? <></>:<LogOut />}
      </div>
    </nav>
  
}
const mapStateToProps = ({ authedUser}) => ({
    authedUser,
  });

export default connect(mapStateToProps)(Nav);
