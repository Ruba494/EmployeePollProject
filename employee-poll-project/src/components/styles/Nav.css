nav {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 0.5rem;
}

nav ul {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  font-size: 16px;
  align-items: center;
  margin: 5px;
}

nav ul li {
  list-style-type: none;
  padding: 15px;
  text-decoration: none;
}
a {
  text-decoration: none;
  color: black;
}


