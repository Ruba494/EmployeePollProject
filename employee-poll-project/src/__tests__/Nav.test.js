import React from "react";
import renderer from "react-test-renderer";
import store from "../utils/test-util";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import Nav from "../components/Nav";

it("should render an empty message when no Nav", () => {
  const elem = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
