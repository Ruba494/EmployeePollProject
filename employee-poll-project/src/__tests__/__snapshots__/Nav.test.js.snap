// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an empty message when no Nav 1`] = `
<nav
  className="nav"
>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/leaderboard"
        onClick={[Function]}
      >
        Lead Board
      </a>
    </li>
    <li>
      <a
        href="/add"
        onClick={[Function]}
      >
        New Poll
      </a>
    </li>
  </ul>
  <div />
</nav>
`;

exports[`should render an empty message when no books 1`] = `
<nav
  className="nav"
>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/leaderboard"
        onClick={[Function]}
      >
        Lead Board
      </a>
    </li>
    <li>
      <a
        href="/add"
        onClick={[Function]}
      >
        New Poll
      </a>
    </li>
  </ul>
  <div />
</nav>
`;
